module/proto/gravity/v1/msgs.proto

```diff
@@ -1,237 +1,176 @@
-// claimed to have seen the deposit enter the ethereum blockchain coins are
-// issued to the Cosmos address in question
-// -------------
-message MsgDepositClaim {
-  uint64 event_nonce    = 1;
-  uint64 block_height   = 2;
-  string token_contract = 3;
-  string amount         = 4 [
-    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Int",
-    (gogoproto.nullable)   = false
+// SignerSetUpdateTxSignature is a signature on behalf of a validator for a SignerSetUpdateTx
+message SignerSetUpdateTxSignature {
+  uint64 nonce                = 1;
+  string eth_signer           = 2;
+  bytes  signature            = 3 [
+    (gogoproto.casttype) = "github.com/ethereum/go-ethereum/common/hexutil.Bytes"
   ];
-  string ethereum_sender = 5;
-  string cosmos_receiver = 6;
-  string orchestrator    = 7;
 }
 
-message MsgDepositClaimResponse {}
+message MsgSubmitEthereumSignatureResponse {}
+
+// MsgSubmitEthereumEvent
+message MsgSubmitEthereumEvent {
+  option (gogoproto.goproto_getters) = false;
 
-// WithdrawClaim claims that a batch of withdrawal
-// operations on the bridge contract was executed.
-message MsgWithdrawClaim {
-  uint64 event_nonce    = 1;
-  uint64 block_height   = 2;
-  uint64 batch_nonce    = 3;
-  string token_contract = 4;
-  string orchestrator   = 5;
+  google.protobuf.Any event  = 1 [(cosmos_proto.accepts_interface) = "EthereumEvent"];
+  string              signer = 2;
 }
 
-message MsgWithdrawClaimResponse {}
-
-// ERC20DeployedClaim allows the Cosmos module
-// to learn about an ERC20 that someone deployed
-// to represent a Cosmos asset
-message MsgERC20DeployedClaim {
-  uint64 event_nonce    = 1;
-  uint64 block_height   = 2;
-  string cosmos_denom   = 3;
-  string token_contract = 4;
-  string name           = 5;
-  string symbol         = 6;
-  uint64 decimals       = 7;
-  string orchestrator   = 8;
+// DepositEvent is submitted when the DepositEvent is emitted by they gravity contract. 
+// ERC20 representation coins are minted to the cosmosreceiver address.
+message DepositEvent {
+  uint64 event_nonce     = 1;
+  string token_contract  = 2;
+  string amount          = 3 [
+    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Int", 
+    (gogoproto.nullable) = false
+  ];
+  string ethereum_sender = 4;
+  string cosmos_receiver = 5;
+  uint64 ethereum_height = 6;
+}
+
+// WithdrawEvent claims that a batch of withdrawal operations on the bridge contract
+// was executed successfully on ETH
+message WithdrawEvent {
+  string token_contract  = 1;
+  uint64 event_nonce     = 2;
+  uint64 ethereum_height = 3;
 }
 
-message MsgERC20DeployedClaimResponse {}
+// ContractCallTxExecutedEvent describes a contract call that has been
+// successfully executed on Ethereum.
 
-// This informs the Cosmos module that a logic
-// call has been executed
-message MsgLogicCallExecutedClaim {
+// NOTE: bytes.HexBytes is supposed to "help" with json encoding/decoding investigate?
```

Comment:
I don't remember who put it in, but I believe it is just a byte string that automatically is serialized into hex.

Reply:
Yeah, seemed to work before. I don't know if I see a reason behind it. Will ask the encoding experts and open an issue if we decide to go ahead with this.